src/VisualStudio/Core/Def/Implementation/Preview/PreviewUpdater.cs

```diff
@@ -96,7 +125,9 @@ private void ApplyDocumentToBuffer(Document document, SpanChange spanSource, out
             }
 
             container = TextView.TextBuffer.AsTextContainer();
-            documentBackedByTextBuffer = document.WithText(container.CurrentText);
+            documentBackedByTextBuffer = isAdditionalDocument ?
+                document.WithAdditionalDocumentText(container.CurrentText) :
+                document.WithText(container.CurrentText);
```

Comment:
this seem can be avoided as well. what is hierarchy between additional and document/textdocument look like?